lib/srv/authhandlers.go

```diff
@@ -407,9 +421,15 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 	// check if the user has permission to log into the node.
 	switch {
 	case h.c.Component == teleport.ComponentForwardingNode:
-		err = h.canLoginWithoutRBAC(cert, clusterName.GetClusterName(), teleportUser, conn.User())
+		// if the target node exists and it is an agentless node,
+		// preform an RBAC check
+		if h.c.TargetNode != nil && h.c.TargetNode.GetSubKind() == types.KindOpenSSHNode {
```

Comment:
Shouldn't we return an error when `h.c.TargetNode == nil && h.c.TargetNode.GetSubKind() == types.KindOpenSSHNode` instead of allowing access without RBAC?

Also, how this case differs from the default case `h.c.Component != teleport.ComponentForwardingNode`?

Reply:
The first case is impossible if `.c.TargetNode == nil`, did you mean `.c.TargetNode != nil && h.c.TargetNode.GetSubKind() != types.KindOpenSSHNode`?

If `h.c.TargetNode` is `nil`, the node is not registered so we can't preform a RBAC check.
If the `AuthHandlers` are running for a regular SSH node, a RBAC check should always be preformed.

Reply:
NVM